pglookout/common.py

```diff
@@ -4,40 +4,44 @@
 Copyright (c) 2015 Ohmu Ltd
 See LICENSE for details
 """
-import datetime
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from typing import Final
+
 import re
 
 
-def convert_xlog_location_to_offset(wal_location):
+def convert_xlog_location_to_offset(wal_location: str) -> int:
     log_id, offset = wal_location.split("/")
     return int(log_id, 16) << 32 | int(offset, 16)
 
 
-ISO_EXT_RE = re.compile(
+ISO_EXT_RE: Final[re.Pattern[str]] = re.compile(
     r"(?P<year>\d{4})-(?P<month>\d\d)-(?P<day>\d\d)(T(?P<hour>\d\d):(?P<minute>\d\d)"
     r"(:(?P<second>\d\d)(.(?P<microsecond>\d{6}))?)?Z)?$"
 )
-ISO_BASIC_RE = re.compile(
+ISO_BASIC_RE: Final[re.Pattern[str]] = re.compile(
     r"(?P<year>\d{4})(?P<month>\d\d)(?P<day>\d\d)(T(?P<hour>\d\d)(?P<minute>\d\d)"
     r"((?P<second>\d\d)((?P<microsecond>\d{6}))?)?Z)?$"
 )
 
 
-def parse_iso_datetime(value):
+def parse_iso_datetime(value: str) -> datetime:
     match = ISO_EXT_RE.match(value)
     if not match:
         match = ISO_BASIC_RE.match(value)
     if not match:
         raise ValueError(f"Invalid ISO timestamp {value!r}")
-    parts = dict(
-        (key, int(match.group(key) or "0")) for key in ("year", "month", "day", "hour", "minute", "second", "microsecond")
-    )
-    return datetime.datetime(tzinfo=None, **parts)
+    parts: dict[str, int] = {
```

Comment:
Is this hint really needed ? It looks like mypy should be able to infer it.

Reply:
You're right. In an earlier version (keeping the `dict()` builtin) it was necessary. But using the `{}` syntax was already solving the issue, and I forgot to check without.
I pushed a new version that should have improved a few things :-)